Recursive generic fn does not work in some cases

```v
fn encode<T>(data T) string {
	return encode_no_len<T>(data)
}

fn encode_no_len<T>(data T) string {
	$if T is u64 {
		return 'u64: $data.str()'
	}
	$if T is string {
		return 'string: $data'
	}
	return 'unknown: $data.str()'
}

fn main() {
	assert encode(u64(43)) == 'u64: 43'
	assert encode('test') == 'string: test'
}
```

Result:
```v
/tmp/v/test.17234021707148904628.tmp.c:9399: warning: implicit declaration of function 'main__encode_no_len_T_string'
/tmp/v/test.17234021707148904628.tmp.c:9399: error: '{' expected (got ";")
```

What is happening is that `encode_no_len` is not being generated for the second type (testing with a bunch of different types In different orders shows it always the second unique type, e.g. if the first two types are the same, the third one is not generated)

The behaviour is identical before and after 8cb44ed9d

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1929,7 +1929,7 @@ pub fn (mut c Checker) fn_call(mut call_expr ast.CallExpr) ast.Type {
 		}
 	}
 	if has_generic_generic {
-		if c.mod != '' {
+		if c.mod != '' && !fn_name.starts_with('${c.mod}.') {
 			// Need to prepend the module when adding a generic type to a function
 			c.table.register_fn_generic_types(c.mod + '.' + fn_name, generic_types)
 		} else {

--- a/vlib/v/tests/generics_in_generics_test.v
+++ b/vlib/v/tests/generics_in_generics_test.v
@@ -1,0 +1,12 @@
+fn gen_fn2<T>(e T) string {
+	return '$e.str()'
+}
+
+fn gen_fn<T>(e T) string {
+	return gen_fn2<T>(e)
+}
+
+fn test_generics_in_generics() {
+	assert gen_fn(u64(42)) == '42'
+	assert gen_fn('42') == '42'
+}

